flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractInput.java

```diff
@@ -68,4 +68,13 @@ public void processLatencyMarker(LatencyMarker latencyMarker) throws Exception {
 	public void setKeyContextElement(StreamRecord record) throws Exception {
 		owner.internalSetKeyContextElement(record, stateKeySelector);
 	}
+
+	@Override
+	public void endInput() throws Exception {
+		if (owner instanceof BoundedOneInput && inputId == 1) {
```

Comment:
Here can it be any other inputId apart from 1?

Reply:
It shouldn't. Nevertheless I think it is not safe to just assume so.

Reply:
The reason I am asking is because this check did not exist in any of the methods removed by this PR which were handling the input closing logic. 

In my opinion, although from the code this check may be safe, to be on the safe side it may make sense to remove it so that the logic is the same. But feel free to disagree and leave it as is.

Reply:
Is it a bug? 

Do we enforce anywhere that a multi input operator cannot implement the `BoundedOneInput`? What if user implements:

```
class CustomOperator extends AbstractStreamOperatorV2 implements BoundedOneInput, BoundedMultiInput
```
?

Reply:
BTW, shouldn't the `BoundedMultiInput` extend from `BoundedOneInput`?

```
interface BoundedMultiInput extends BoundedOneInput {
 
        default void endInput() throws Exception {
            endInput(1)
        }

	/**
	 * It is notified that no more data will arrive on the input identified by the {@code inputId}.
	 * The {@code inputId} is numbered starting from 1, and `1` indicates the first input.
	 */
	void endInput(int inputId) throws Exception;
}
```

Reply:
I think this could lead to more confusion and bugs. It would be better to keep it simple and not support it. `endInput()` in your example above someone could understand as ending all inputs. 

Also as always. Keeping it simpler now, means we can always extend it in the future if we change our mind. But if we decide to support some  strange combination now "just because we can", backing out from it won't be possible without braking compatibility.

Reply:
If you say so. 

BTW, I think this discussion is somewhat unrelated to the PR, but ...
I find the contract between `BoundedOneInput` and `BoundMultiInput` undefined/ill defined.

Are these combinations valid? In the current implementation
1. `TwoInputOperator implements BoundedOneInput` -> whenver any input finishes, we call BoundedOneInput#endInput, imo this is confusing
2. `OneInputOperator implements BoundedMultiInput` -> we call BoundedMultiInput#endInput(1) <- sort of ok, imo
3. `MultiInputOperator implements BoundedOneInput` -> see 1.
4. `MultiInputOperator implements BoundedMultiInput, BoundedOneInput` -> we only ever call BoundedOneInput#endInput for all finished inputs, which imo is misleading

I know that this PR, does not fixes it. In the PR I do ignore options 1 and 2 though.

Reply:
You are right, it's currently an undefined behaviour. Imo all of the above should be disallowed.